net/http/client.cpp

```diff
@@ -224,9 +224,9 @@ class ClientImpl : public Client {
             LOG_ERROR_RETURN(EINVAL, ROUNDTRIP_FAILED,
                             "Content-Length and Transfer-Encoding conflicted");
         }
-        op->req.headers.insert("User-Agent", USERAGENT);
-        op->req.headers.insert("Connection", "keep-alive");
         op->req.headers.merge(m_common_headers);
+        op->req.headers.insert("User-Agent", m_user_agent.empty() ? USERAGENT : m_user_agent);
```

Comment:
m_user_agent.empty() ? string_view(USERAGENT) : string_view(m_user_agent)

Reply:
otherwise, USERAGENT may be converted (constructed) to std::string, then string_view